frontend/app/module/input/PollingStation/PollingStation.test.tsx

```diff
@@ -0,0 +1,325 @@
+      expectRecountedForm,
+      fillRecountedForm,
+      submit,
+      expectVotersAndVotesForm,
+      fillVotersAndVotesForm,
+      submit,
+      expectDifferencesForm,
+      fillDifferencesForm,
+      submit,
+      // navigate back to the 'voters and votes' form
+      () => userEvent.click(screen.getByRole("link", { name: "Aantal kiezers en stemmen" })),
+      expectVotersAndVotesForm,
+    ];
+
+    for (const step of steps) {
+      await step();
+    }
+  });
+
+  test("Navigate back with dirty state", async () => {
+    render();
+
+    const steps = [
+      // fill up to and including the voters and votes form
+      startPollingStationInput,
+      expectRecountedForm,
+      fillRecountedForm,
+      submit,
+      expectVotersAndVotesForm,
+      fillVotersAndVotesForm,
+      // don't submit, and navigate back to the 'recounted' form
+      () => userEvent.click(screen.getByRole("link", { name: "Is er herteld?" })),
+      expectRecountedForm,
+    ];
+
+    for (const step of steps) {
+      await step();
+    }
+  });
+
+  test("Navigate to correct page after navigating back and submitting", async () => {
+    render();
+
+    const steps = [
+      // fill up to and including the differences form
+      startPollingStationInput,
+      expectRecountedForm,
+      fillRecountedForm,
+      submit,
+      expectVotersAndVotesForm,
+      fillVotersAndVotesForm,
+      submit,
+      expectDifferencesForm,
+      fillDifferencesForm,
+      submit,
+      // navigate back to the 'voters and votes' form and submit
+      () => userEvent.click(screen.getByRole("link", { name: "Aantal kiezers en stemmen" })),
+      expectVotersAndVotesForm,
+      submit,
+      // expect to be on the 'differences' form
+      expectDifferencesForm,
+    ];
+
+    for (const step of steps) {
+      await step();
+    }
+  });
+
+  test("Navigate triggers changes modal", async () => {
+    render();
+    await startPollingStationInput();
+    await expectRecountedForm();
+    await fillRecountedForm();
+    await submit();
+
+    await expectVotersAndVotesForm();
+
+    await fillVotersAndVotesForm();
+    await submit();
+
+    await expectDifferencesForm();
+
+    await userEvent.click(screen.getByRole("link", { name: "Aantal kiezers en stemmen" }));
+
+    await expectVotersAndVotesForm();
+
+    await fillVotersAndVotesForm({
+      poll_card_count: 1,
+      proxy_certificate_count: 1,
+      voter_card_count: 1,
+      total_admitted_voters_count: 2,
+      votes_candidates_counts: 1,
+      blank_votes_count: 1,
+      invalid_votes_count: 1,
+      total_votes_cast_count: 3,
+    });
+
+    await submit();
+    await expectFeedbackError();
+
+    await fillVotersAndVotesForm({
+      total_admitted_voters_count: 3,
+    });
+
+    await userEvent.click(screen.getByRole("link", { name: "Verschillen" }));
+
+    await expectBlockerModal();
+  });
+
+  test("Progress list shows correct icons", async () => {
```

Comment:
Am missing the icons "nog niet afgerond", "bevat een fout", "je bent hier".
Also see https://github.com/kiesraad/abacus/blob/133.frontend.update-data-entry-navigation/frontend/e2e-tests/dom-to-db-tests/data-entry.e2e.ts#L352

Reply:
added checks